From 988d9e649b43bad46aeb3b69ffc2ca5b15902ee9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Men=C5=A1=C3=ADk?= <pemensik@redhat.com>
Date: Sun, 12 Mar 2023 13:04:26 +0100
Subject: [PATCH] Provide simple helper for cJSON location

If BUNDLED_LIBS is disabled and cjson library does not ship its cmake
config for some reason, attempt to find it the other way. That is the
case on current Fedora cjson package.

Provide CJSON_LIBRARY in addition to CJSON_LIBRARIES variable.
---
 cmake/FindcJSON.cmake | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)
 create mode 100644 cmake/FindcJSON.cmake

diff --git a/cmake/FindcJSON.cmake b/cmake/FindcJSON.cmake
new file mode 100644
index 000000000..ef28a8dc0
--- /dev/null
+++ b/cmake/FindcJSON.cmake
@@ -0,0 +1,20 @@
+# Simple wrapper for locating cJSON package even if cmake config is not available
+find_package(PkgConfig)
+if (PKG_CONFIG_FOUND)
+	pkg_check_modules(CJSON libcjson)
+	set (CJSON_LIBRARY ${CJSON_LIBRARIES})
+endif()
+
+if (NOT CJSON_FOUND)
+	find_path(CJSON_INCLUDE_DIRS NAMES cjson.h PATH_SUFFIXES cjson)
+	find_library(CJSON_LIBRARIES NAMES cjson)
+
+	if (CJSON_INCLUDE_DIRS and CJSON_LIBRARIES)
+		message(STATUS "Found cJSON: ${CJSON_LIBRARIES}")
+		set (CJSON_FOUND TRUE)
+		# Should this one be name like all other variables too?
+		set (CJSON_LIBRARY ${CJSON_LIBRARIES})
+	endif()
+endif()
+
+MARK_AS_ADVANCED(CJSON_INCLUDE_DIRS CJSON_LIBRARIES CJSON_LIBRARY)
-- 
2.39.2

